### PR TITLE
pkg/kmodule: add an interface type so we can plan to use on non-Linux…

### PR DESCRIPTION
**Pull request by rminnich**
… systems

Also add a Darwin implementation; we now have lsmod on Darwin, with more to come later.

The global Probe and other functions must remain, since they are referred to elsewhere.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
If this one looks reasonable I'll fix the coverage.

File: pkg/cmdline/cmdline_test.go [hunk 1]
```diff
@@ -85,15 +85,12 @@ func TestCmdlineModules(t *testing.T) {
 func TestCmdLineClassic(t *testing.T) {
 	c := getCmdLine()
 	if c.Err != nil {
-		t.Skipf("getCmdLine(): got %v, want nil, skipping test", c.Err)
+		t.Skipf("getCmdLine(): got %v, want nil, proably a non-modular kernel, skipping test", c.Err)
```
Comment (abrender) [inline pkg/cmdline/cmdline_test.go:88]: typo: `proably`

**Review by abrender — APPROVED** [review]

**Comment by rminnich** [comment]
n.b.: don't squash commit. There is one change I want to be able to bisect around. It looks right, but with modules, caution is the order of the day.

**Review by ChriMarMe — COMMENTED** [review]
The first commit is a behemoth, while the second and third are virtually empty. Also concerning that the commit messages and the content does not match too well in second and third commit.

File: pkg/kmodule/kmodule_linux_test.go [hunk 2]
```diff
@@ -151,3 +151,13 @@ func TestFileInit(t *testing.T) {
...
 }
+
+func TestLoadModule(t *testing.T) {
+	l, err := NewPath("proc.modules")
+	if err != nil {
+		t.Fatalf(`NewPath("proc.modules"): %v != nil`, err)
+	}
+	// This is not going to work in CI, but at least it covers
+	// more code.
+	l.loadModule("data/cpuid.ko", "", ProbeOpts{})
```
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/kmodule/kmodule_linux_test.go:162]: The CI does not use a modular kernel. We need to rebuild all the kernels in CI before this will work.

**Comment by rminnich** [comment]
I'll do this another time, it seems just too big.

**Pull request closed without merging**